.circleci/config.yml

```diff
@@ -0,0 +1,307 @@
+
+  static_analysis:
+    <<: *container_config
+    steps:
+      - checkout
+      - *restore_rebar_cache
+      - *restore_build_cache
+      - run: ./rebar3 edoc
+      - restore_cache:
+          keys:
+            - dialyzer-cache-v2-{{ .Branch }}-{{ .Revision }}
+            - dialyzer-cache-v2-{{ .Branch }}-
+            - dialyzer-cache-v2-
+      - run:
+          name: Update dialyzer PLT
+          command: make dialyzer-install
+      - save_cache:
+          key: dialyzer-cache-v2-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - _build/default/rebar3_20.2.2_plt
+      - run: make dialyzer
+
+  linux_package:
+    <<: *container_config
+    steps:
+      - checkout
+      - run: id
+      - *build_package
+      - *test_package
+      - *store_package_artifacts
+      - persist_to_workspace:
+          root: *packages_workspace
+          paths:
+            - "*.tar.gz"
+
+  osx_package:
+    macos:
+      xcode: "9.0"
+    working_directory: ~/epoch
+    steps:
+      - checkout
+      - run:
+          name: Install required tools
+          command: |
+            brew update
+            brew install erlang
+      - *build_package
+      - *test_package
+      - *store_package_artifacts
+
+  deploy_integration:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: integration
+    steps: *deploy_steps
+
+  deploy_dev1:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev1
+    steps: *deploy_steps
+
+  deploy_dev2:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev2
+    steps: *deploy_steps
+
+  deploy_uat:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: uat
+    steps: *deploy_steps
+
+# CircleCI skips a job for a tag by default.
+# A job must have a filters tags section to run as a part of a tag push
+# and all its transitively dependent jobs must also have a filters tags section.
+workflows:
+  version: 2
+  build_test_deploy:
+    jobs:
+      - build:
+          requires: ~
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - test:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - eunit:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - static_analysis:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - linux_package:
+          filters:
+            branches:
+              only:
+                - master
```

Comment:
I see packages (linux and osx) built and smoke tested only on master - not on PRs/pushes. How does developer know that he/she broke package smoke test? I am concerned failures on master may go silent and package smoke test issue noticed only on tag.

Reply:
I've never seen failed "package" job on travis so far, that's why I was convinced to propose to take out it from the PR workflow and keep it only for master.

Furthermore we also should work to make sure that failed master is much more visible than now (e.g. slack notification *but when we migrate to paid plan, because current slack hooks are exhausted) and fixed with the highest priority.

I believe this would overall improve the PR build times with the cost of very small risk.